### PR TITLE
Catch the illegal argument exception in Net Framework!

### PR DESCRIPTION
**Pull request by JaynieBai**
Fixes [#8762 ](https://github.com/dotnet/msbuild/issues/8762)

### Context
Catch the exceptions when extensionsPathPropValue is null or importExpandedWithDefaultPath is empty.
In NET Framework, Path.* function also throws exceptions if the path contains invalid characters

### Changes Made
Catch the exception.

### Testing
FallbackImportWithInvalidProjectValue

### CODE REVIEW

File: src/Build/Evaluation/Evaluator.cs [hunk 1]
```diff
@@ -2547,7 +2547,16 @@ private void ThrowForImportedProjectWithSearchPathsNotFound(ProjectImportPathMat
                         importElement.Project.Replace(searchPathMatch.MsBuildPropertyFormat, extensionsPathPropValue),
                         ExpanderOptions.ExpandProperties, importElement.ProjectLocation);
 
-                relativeProjectPath = FileUtilities.MakeRelative(extensionsPathPropValue, importExpandedWithDefaultPath);
+                try
+                {
+                    relativeProjectPath = FileUtilities.MakeRelative(extensionsPathPropValue, importExpandedWithDefaultPath);
+                }
+                catch (ArgumentException)
```
Comment (JanKrivanek) [inline src/Build/Evaluation/Evaluator.cs:2554]: There can be multiple reasons for ArgumentException in here - let's not loose that information. Either add it to the error message, or (probably better) - propagate it to the `InvalidProjectFileException` InnerException.

Comment (JaynieBai) [reply src/Build/Evaluation/Evaluator.cs]: Done

**Review by JanKrivanek — APPROVED** [review]
Looks good!

File: src/Build/Evaluation/Evaluator.cs [hunk 2]
```diff
@@ -2547,7 +2547,16 @@ private void ThrowForImportedProjectWithSearchPathsNotFound(ProjectImportPathMat
                         importElement.Project.Replace(searchPathMatch.MsBuildPropertyFormat, extensionsPathPropValue),
                         ExpanderOptions.ExpandProperties, importElement.ProjectLocation);
 
-                relativeProjectPath = FileUtilities.MakeRelative(extensionsPathPropValue, importExpandedWithDefaultPath);
+                try
+                {
+                    relativeProjectPath = FileUtilities.MakeRelative(extensionsPathPropValue, importExpandedWithDefaultPath);
+                }
+                catch (ArgumentException ex)
```
Comment (rainersigwald) [inline src/Build/Evaluation/Evaluator.cs:2554]: This is a good approach to work around the .NET Framework behavior, but I wonder if we can instead take advantage of the .NET Core behavior even in .NET Framework, using the `Microsoft.IO.Redist` package. Can you please try rewriting `FileUtilities.MakeRelative` to use `Microsoft.IO.Path.GetRelativePath` on .NET Framework, which shouldn't throw and should also be faster?

Comment (JaynieBai) [reply src/Build/Evaluation/Evaluator.cs:2554]: @rainersigwald Thanks for your suggestions. But Path.GetRelativePath still throw exceptions when the path is empty.

Comment (JaynieBai) [reply src/Build/Evaluation/Evaluator.cs:2554]: @rainersigwald  This fix is not only for .Net Framework.   Both .Net and .NET Framework, has the exception when extensionsPathPropValue is null or importExpandedWithDefaultPath is empty.  You can see the this from the test.

**Review by YuliiaKovalova — APPROVED** [review]

File: src/Build.UnitTests/Evaluation/ImportFromMSBuildExtensionsPath_Tests.cs [hunk 3]
```diff
@@ -842,6 +843,42 @@ public void FallbackImportWithFileNotFoundWhenPropertyNotDefined()
...
+                var logger = new MockLogger();
+                projectCollection.RegisterLogger(logger);
+
+                Assert.Throws<InvalidProjectFileException>(() => projectCollection.LoadProject(mainProjectPath));
+                logger.AssertLogContains("MSB4102");
+            }
+            finally
+            {
+                FileUtilities.DeleteNoThrow(mainProjectPath);
+            }
```
[the remark this answers is not in the document]
Comment (JaynieBai) [reply src/Build.UnitTests/Evaluation/ImportFromMSBuildExtensionsPath_Tests.cs]: Updated

**Pull request merged**